custom/icds/models.py

```diff
@@ -0,0 +1,163 @@
+        """
+        if not isinstance(case, CommCareCase):
+            case = CommCareCase.wrap(case)
+        self.case = case
+        self.snapshot_start_date = transform_date(snapshot_start_date)
+        self.snapshot_end_date = transform_date(snapshot_end_date)
+        self.forms_provider = forms_provider
+
+    # needs caching across UCR build cycle
+    @property
+    @memoized
+    def forms(self):
+        if self.forms_provider:
+            return self.forms_provider
+        forms = []
+        for form_id in self.case.xform_ids:
+            forms.append(FormAccessors(domain).get_form(form_id))
+
+    def case_property(self, case, property, default=None):
+        prop = case.dynamic_case_properties().get(property, default)
+        if isinstance(prop, basestring) and prop.strip() == "":
+            return default
+        return prop
+
+    def filtered_forms(self, xmlns):
+        for form in self.forms:
+            if form.xmlns == xmlns and are_in_same_month(form.received_on, self.snapshot_start_date):
+                yield form
+
+    @property
+    @memoized
+    def is_open(self):
+        return are_in_same_month(self.case.opened_on, self.snapshot_start_date)
+
+    @property
+    @memoized
+    def nutrition_status(self):
+        statuses = [
+            form.get_data("form/z_score_wfa_ql/nutrition_status")
+            for form in self.filtered_forms(GMP_XMLNS)
+            if form.get_data("form/z_score_wfa_ql/nutrition_status")
+        ]
+        return statuses[0] if any(statuses) else 'unweighed'
+
+    @property
+    @memoized
+    def dob(self):
+        return transform_date(self.case_property(self.mother_case, "dob"))
+
+    @property
+    @memoized
+    def age_in_days(self):
+        return (self.dob - self.snapshot_end_date).days
+
+    @property
+    @memoized
+    def age_in_months(self):
+        return self.age_in_days / 30.4
+
+    @property
+    @memoized
+    def age_in_years(self):
+        return self.age_in_days / 365.25
+
+    @property
+    @memoized
+    def age_group(self):
+        if self.age_in_days <= 28:
+            return '0-28days'
+        elif self.age_in_days > 28 and self.age_in_months <= 6:
+            return '28-6mo'
+        elif 6 < self.age_in_months <= 12:
+            '6-12mo'
+        elif 12 < self.age_in_months <= 24:
+            '1-2yr'
+        elif 24 < self.age_in_months <= 36:
+            '2-3yr'
+        elif 36 < self.age_in_months <= 48:
+            '3-4yr'
+        elif 48 < self.age_in_months <= 60:
+            '4-5yr'
+        elif 60 < self.age_in_months <= 72:
+            '5-6yr'
+
+    @property
+    @memoized
+    def mother_case(self):
+        return self.case.parent
+
+    @property
+    @memoized
+    def household_case(self):
+            return self.mother_case.parent if self.mother_case else None
+
+    def nutrition_status_severely_underweight(self):
+        return one_or_zero(self.nutrition_status == 'severely underweight')
```

Comment:
it looks like a lot of these could be computed in stock UCR - why are they being added here?